### DISCUSSION TITLE
break on element if at the bottom

### DISCUSSION THREAD

**Discussion by liedekef** [dompdf/dompdf#3165] [Q&A] [answered]
Sometimes I have specific elements at the bottom of a page (like a h2 for example, or the title of a section where I want to keep the rest of the text together with the title). Currently it can happen that the element-title is shown at the bottom of a page and the rest at the next page.
Is it possible to include some logic/css to say that 2 elements belong together and so do a page-split before the first element if the first element is at the bottom of a page?

**Comment by bsweeney** [comment, marked as answer]
Probably the easiest way would be to style the H2 with `page-break-after: avoid;`. The alternate method would be to wrap the related elements in a DIV styled with `page-break-inside: avoid;`

**Reply by Mellthas** [reply]
Well, seems we replied at the same time :-D.

**Comment by Mellthas** [comment]
You can control page breaks with the CSS properties `page-break-before`, `page-break-after`, and `page-break-inside`. E.g. to ensure there is no page break directly after a heading, you could use:

```css
h2 {
    page-break-after: avoid;
}
```

or if it that should only apply when the heading is followed by a specific element:

```css
h2 + .class {
    page-break-before: avoid;
}
```

**Comment by liedekef** [comment]
Thank you both. I can only mark one as answer though :-D
